ffi/diplomat/src/collator.rs

```diff
@@ -0,0 +1,239 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatResult;
+    use icu_collator::{Collator, CollatorOptions};
+
+    use crate::{
+        errors::ffi::ICU4XError, locale::ffi::ICU4XLocale, provider::ffi::ICU4XDataProvider,
+    };
+
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu_collator::Collator, Struct)]
+    pub struct ICU4XCollator(pub Collator);
+
+    #[diplomat::enum_convert(core::cmp::Ordering)]
+    #[diplomat::rust_link(core::cmp::Ordering, Enum)]
+    pub enum ICU4XOrdering {
+        Less = 0,
+        Equal = 1,
+        Greater = 2,
+    }
```

Comment:
Suggestion: Should this go in a more general place? Maybe `common.rs`?

I could even see an argument for it to go into diplomat runtime, but we shouldn't hold on that.

Reply:
Elango asked me about this yesterday, I said that the file structure of ffi/diplomat does not matter except for the icu_capi docs, which have limited utility, so probably keeping this in collator.rs for now is fine and we can move it around later if we need)